xbmc/platform/PlatformFactory.cpp

```diff
@@ -0,0 +1,40 @@
+/*
+ *      Copyright (C) 2016 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Kodi; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "PlatformFactory.h"
+#include "interfaces/IPlatform.h"
+#if defined(TARGET_DARWIN)
+#include "platform/darwin/PlatformDarwin.h"
+#endif
+
+IPlatform *CPlatformFactory::GetPlatform()
+{
+#if defined(TARGET_DARWIN)
```

Comment:
Mhhh yes thats neat. But this depends on CORE_SYSTEM_NAME ... there is CORE_SYSTEM_NAME osx and ios ... but when doing the ifdef stuff i have TARGET_DARWIN, TARGET_DARWIN_OSX, TARGET_DARWIN_IOS (and TARGET_DARWIN_TVOS...) - i make use of TARGET_DARWIN here because it applies to all of them. With the linked approach i would need to have the same file there multiple times (one for osx and one for ios).